gio/src/file_enumerator.rs

```diff
@@ -16,3 +17,124 @@ impl Iterator for FileEnumerator {
 }
 
 impl FusedIterator for FileEnumerator {}
+
+pub trait FileEnumeratorExtManual {
+    fn into_stream(self, num_files: i32, priority: glib::Priority) -> FileEnumeratorStream;
+}
+
+impl<O: IsA<FileEnumerator>> FileEnumeratorExtManual for O {
+    // rustdoc-stripper-ignore-next
+    /// Converts the enumerator into a [`Stream`](futures_core::Stream).
+    fn into_stream(self, num_files: i32, priority: glib::Priority) -> FileEnumeratorStream {
+        let future = Some(self.next_files_future(num_files, priority));
+        FileEnumeratorStream {
+            enumerator: self.upcast(),
+            future,
+            num_files,
+            priority,
+        }
+    }
+}
+
+// rustdoc-stripper-ignore-next
+/// A [`Stream`](futures_core::Stream) used to enumerate files in directories.
+pub struct FileEnumeratorStream {
+    enumerator: FileEnumerator,
+    future: Option<LocalBoxFuture<'static, Result<Vec<FileInfo>, glib::Error>>>,
+    num_files: i32,
+    priority: glib::Priority,
+}
+
+impl std::fmt::Debug for FileEnumeratorStream {
+    #[inline]
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("FileEnumeratorStream")
+            .field("enumerator", &self.enumerator)
+            .field("num_files", &self.num_files)
+            .field("priority", &self.priority)
+            .finish()
+    }
+}
+
+impl futures_core::Stream for FileEnumeratorStream {
+    type Item = Result<Vec<FileInfo>, glib::Error>;
```

Comment:
Why not a `glib::List<FileInfo>`?

Reply:
tbh, I'd leave `glib::List` to API that interface directly with C. This is a Rust convenience, so I'd recommend using native/idiomatic Rust type wherever possible.

Reply:
`glib::List` works as an `Iterator` so that for the majority of uses it should be equivalent except for not having to copy over everything into a `Vec`.

Reply:
This is just passing the vec from the underlying `next_files_future`, it is already converted by the time the stream gets it. It should probably be changed in gir to use `List`, and then it can be changed here

Reply:
See https://github.com/gtk-rs/gir/pull/1447